### PR TITLE
Initial support for tags

### PR DESCRIPTION
**Pull request by zquestz**
I have added Tags() to the provider interface so you can easily launch multiple searches at the same time. Figured it is useful when trying to search for movies, news, etc.

- [x] Updated all auto complete scripts to include the new flags.
- [x] Supports searching a tag and a specific provider. Useful if you want to search a tag, but also google.
- [x] Supports adding tags to custom providers. Just add "tags" to your config.
- [x] Added `--list-tags` flag to display all available tags.
- [x] Added Tags() method to all providers.
- [x] Updated README.md
- [x] Added support for tag expansion.
- [x] Populate all providers with an initial set of tags.
- [x] Update -v flag to support list views.

### CODE REVIEW

**Comment by akb** [comment]
Idea: environment conditional tags (e.g. only show arch stuff if using arch)

"photos": 500px, 
"streaming": 8tracks bandcamp
"music": 8tracks bandcamp
"shopping": amazon
"software": archpkg atmospherejs aur codepen
"arch": archpkg archwiki aur
"info": arxiv
"papers": arxiv
"javascript": atmospherejs codepen
"general": baidu bing
"news": arstechnica bgr buzzfeed cnn
"tech": arstechnica bgr
"bullshit": buzzfeed

**Comment by akb** [comment]
"arch": archpkg archwiki aur
"code": dockerhub github stackoverflow
"education": arxiv coursera libgen wikipedia
"forums": quora reddit
"general": bing duckduckgo google yahoo
"go": go godoc
"javascript": atmospherejs npm
"movies": crunchyroll netflix
"music": 8tracks bandcamp soundcloud spotify
"news": cnn npr theregister
"photos": 500px flickr instagram
"pics": dribbble flickr imgur instagram
"shopping": amazon overstock
"social": facebook googleplus twitter
"tech-news": arstechnica bgr engadget hackernews
"torrent": kickasstorrents thepiratebay torrentz
"video": vimeo youtube

**Comment by zquestz** [comment]
Yeah that looks like a reasonable start. Thanks @akb!

**Comment by shr3kst3r** [comment]
Maybe change "general" to "search".

**Comment by zquestz** [comment]
@shr3kst3r I was thinking the same thing. =)

**Comment by zquestz** [comment]
Probably will change torrent to torrents as well.

**Comment by zquestz** [comment]
What do you guys think of ecom instead of shopoing?

**Comment by zquestz** [comment]
Ok this is now ready for review. I went ahead and added tag expansion. This will search the tag tech-news for ssd info.

```
s -t te ssd
```

I also added the tags as suggested, but changed general to search, and removed dockerhub from code, as I didn't think it fit.

**Comment by zquestz** [comment]
Also went ahead and fixed the list views. Here is sample output:

s -l -v
```
500px (photos)
8tracks (music)
amazon (shopping)
archpkg (arch)
archwiki (arch)
arstechnica (tech-news)
arxiv (education)
atmospherejs (javascript)
aur (arch)
baidu
bandcamp (music)
bgr (tech-news)
bing (search)
buzzfeed
cnn (news)
codepen
coursera (education)
cplusplus
crunchyroll (movies)
debianpkg
digg
diigo
dockerhub
dribbble (pics)
duckduckgo (search)
dumpert
engadget (tech-news)
facebook (social)
flickr (photos, pics)
flipkart
foursquare
gist
github (code)
gmail
go (go)
godoc (go)
google (search)
googledocs
googleplus (social)
hackernews (tech-news)
ietf
ifttt
imdb (movies)
imgur (pics)
inbox
instagram (photos, pics)
kickasstorrents (torrents)
libgen (education)
linkedin
lmgtfy
macports
mdn
metacpan
msdn
naver
netflix (movies)
nhaccuatui
npm (javascript)
npmsearch (javascript)
npr (news)
nvd
overstock (shopping)
packagist
phandroid
php
pinterest
python
quora (forums)
reddit (forums)
rottentomatoes
rubygems
soundcloud (music)
spotify (music)
stackoverflow (code)
steam
taobao
thepiratebay (torrents)
theregister (news)
torrentz (torrents)
twitchtv
twitter (social)
unity3d
upcloud
vimeo (video)
wikipedia (education)
wolframalpha
yahoo (search)
yandex
youtube (video)
```

**Comment by tw4dl** [comment]
rottentomatoes (movies)
gist (code)

**Comment by zquestz** [comment]
@tw4dl thanks, those were adjusted. =)

File: providers/taobao/taobao.go [hunk 1]
```diff
@@ -18,3 +18,8 @@ type Provider struct{}
 func (p *Provider) BuildURI(q string) string {
 	return fmt.Sprintf("https://s.taobao.com/search?q=%s", url.QueryEscape(q))
 }
+
+// Tags returns the tags relevant to this provider.
+func (p *Provider) Tags() []string {
+	return []string{}
```
Comment (lilirui) [inline providers/taobao/taobao.go]: This should be 'shopping' also.

Comment (zquestz) [inline providers/taobao/taobao.go]: Good call, one sec.

**Comment by lilirui** [comment]
taobao (shopping)
baidu (search)

**Comment by zquestz** [comment]
Added =)

**Comment by zquestz** [comment]
@KeizerDev any thoughts?

**Comment by zquestz** [comment]
Going to merge this in about an hour. If anyone has suggestions or changes please let me know!

**Comment by tw4dl** [comment]
yandex (search)
inbox gmail (email)

**Comment by zquestz** [comment]
Not going to do an email tag as most users just use one main mail provider. Yandex was added. =)

**Pull request merged**